src/context.rs

```diff
@@ -57,50 +59,17 @@ pub struct Tree {
 
     /// The current span node. This is the node that is currently being polled.
     current: NodeId,
+
+    /// Whether to coloring the terminal
+    colored: bool,
+
+    /// if the time of execution is beyond it, warn it
+    warn_threshold: time::Duration,
```

Comment:
What about directly storing a `Config` here?

Reply:
> What about directly storing a `Config` here?

If put `Config` here, the field `verbose` is not used by context.

Reply:
Yeah, you're right. So I'm considering whether `colored` and `warn_threshold` should be properties of tree formatting, instead of the tree or the registry. 🤔 For example, we have...

```rust
struct FormatConfig {
  colored: bool,
  warn_threshold: time::Duration,
}

pub struct FormatTree<'a> {
  tree: &'a Tree,
  config: FormatConfig
}

impl Display for FormatTree<'_> { .. }

impl Tree {
  fn display(&self, config: FormatConfig) -> FormatTree<'_> { .. }
}

impl Display for Tree { /* delegate to `self.display(FormatConfig::default()).fmt(f)` */ }
```

Reply:
Yes, It could be done this way.